packages/app/src/request.ts

```diff
@@ -32,17 +37,21 @@ export const getProtocol = (req: Request, trust: Trust): Protocol => {
   return index !== -1 ? header.substring(0, index).trim() : header.trim()
 }
 
-export const getHostname = (req: Request, trust: Trust): string | undefined => {
+export const getHost = (req: Request, trust: Trust): Host | undefined => {
   let host: string = req.get('X-Forwarded-Host') as string
 
   if (!host || !trustRemoteAddress(req, trust)) host = req.get('Host') as string
 
-  if (!host) return
+  if (!host) return undefined
 
   // IPv6 literal support
   const index = host.indexOf(':', host[0] === '[' ? host.indexOf(']') + 1 : 0)
+  if (index === -1) return { hostname: host }
 
-  return index !== -1 ? host.substring(0, index) : host
+  const hostname = host.substring(0, index)
+  const port = Number(host.substring(index + 1))
+  if (Number.isNaN(port)) throw new TypeError('Port number is NaN, therefore Host is malformed')
```

Comment:
Just realized something: it's better to avoid throwing an error cause if during request lifecycle, an invalid host header is sent, the end user's app will crash. Instead, return `null`.

Reply:
I don't think that's true, I'm pretty confident the error will be caught by the top-level error handler, leading to a `500 Internal Server Error` response

Reply:
Can you add a test case for concreteness?